app/lib/services/funding_eligibility.rb

```diff
@@ -14,9 +15,16 @@ def call
            "NPQ for Senior Leadership (NPQSL)",
            "NPQ for Executive Leadership (NPQEL)"
         eligible_establishment_type_codes_for_base_courses.include?(school.establishment_type_code) && school.high_pupil_premium
-      when "NPQ Leading Teacher Development (NPQLTD)",
-           "NPQ for Headship (NPQH)"
-        eligible_establishment_type_codes_for_ltd_or_headship_courses.include?(school.establishment_type_code)
+      when "NPQ Leading Teacher Development (NPQLTD)"
+        eligible_establishment_type_codes_for_ltd_courses.include?(school.establishment_type_code)
+      when "NPQ for Headship (NPQH)"
+        a = if headteacher_status == "yes_in_first_two_years"
+              eligible_establishment_type_codes_for_new_headship.include?(school.establishment_type_code)
+            end
+
+        b = school.high_pupil_premium && eligible_establishment_type_codes_for_headship_and_high_pupil_premiums.include?(school.establishment_type_code)
+
+        a || b
```

Comment:
```suggestion
        in_first_two_years = if headteacher_status == "yes_in_first_two_years"
              eligible_establishment_type_codes_for_new_headship.include?(school.establishment_type_code)
            end

        high_pupil_premium = school.high_pupil_premium && eligible_establishment_type_codes_for_headship_and_high_pupil_premiums.include?(school.establishment_type_code)

        in_first_two_years || high_pupil_premium
```

Reply:
i renamed to something slightly different but same idea